Source/Core/DolphinQt/DiscordJoinRequestDialog.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QDialog>
+#include <ctime>
+
+class QGridLayout;
+class QPixmap;
+
+class DiscordJoinRequestDialog : public QDialog
+{
+  Q_OBJECT
+public:
+  explicit DiscordJoinRequestDialog(QWidget* parent, const char* id, const std::string& discord_tag,
+                                    const char* avatar);
+  std::time_t GetCloseTimestamp() const;
+
+private:
+  void CreateMainLayout(const std::string& discord_tag, const QPixmap& avatar);
```

Comment:
I looked at other files and saw MainLayout being used and used that to keep constant naming. However, I do agree that it doesn't make much sense since there's only one.